### PR TITLE
W43 catch up mailgun

### PR DESCRIPTION
**Pull request by JPPortier**
'resend' end point support

### CODE REVIEW

File: client/src/main/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializer.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Map;
+
+public class ArrayToCommaSeparatedFormSerializer extends FormSerializer<List<String>> {
+
+  @Override
+  public void serialize(List<String> in, String fieldName, Map<String, Object> out) {
+    if (null != in) {
+      out.put(fieldName, String.join(",", in));
+    }
+  }
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializer.java:14]: I just want to be sure about this method: it will concatenate all the list entries and put a comma between them. You have covered in your tests the fact a list entry can contain a comma-separated list of values. 
That's a great edge case to have considered! Would you like to split this kind of entries and trim the data before doing the concatenation?

File: client/src/test/java/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializerTest.java
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializerTest.java:1]: What about the entry `Arrays.asList("value1", "value2,,,,,")`
Should it be serialized into `value1,value2,,,,,` or `value1,value2`

File: sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      LOGGER.severe(e.getMessage());
+      e.printStackTrace();
+    }
+  }
+
+  public void run() {
+
+    EmailsService service = client.mailgun().v1().emails();
+
+    LOGGER.info("ReSend email with Mailgun to: " + mailgunTo);
```
Comment (asein-sinch) [inline sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java:30]: Where is defined this variable?

File: sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    String storageKey = "storage key value";
+
+    ResendRequest parameters = ResendRequest.builder().setTo(Arrays.asList(mailgunTo)).build();
+
+    SendEmailResponse response = service.resend(mailgunDomain, storageKey, parameters);
+
+    LOGGER.info("Response: " + response);
+  }
+
+  SendEmailRequest createTextEmail() {
```
Comment (asein-sinch) [inline sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java]: Why do you need this method in the ReSend class ?

File: client/src/main/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializer.java (hunk 1 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializer.java:14]: If there is such constraint from backend to respect this: yes we could have to implement this as "helper.

But there is no such requirements from backend; then no justification to add dedicated business logic in front of request

File: client/src/test/java/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializerTest.java
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/mailgun/api/v1/adapters/ArrayToCommaSeparatedFormSerializerTest.java:1]: No dedicated (nor required by mailgun API) business logic here; then `value1,value2,,,,,` is expected

Added dedicated tests to "document" it

File: sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java (hunk 2 above)
Comment (JPPortier) [reply sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java:30]: Read from configuration file: https://github.com/sinch/sinch-sdk-java/blob/feat/V2.0-next/sample-app/src/main/resources/config.properties#L18

And available from test because if inherit of `BaseApplication`: https://github.com/sinch/sinch-sdk-java/blob/feat/V2.0-next/sample-app/src/main/java/com/sinch/sample/BaseApplication.java#L64

File: sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java (hunk 3 above)
Comment (JPPortier) [reply sample-app/src/main/java/com/sinch/sample/mailgun/messages/ReSend.java]: 🤦 
Removed.
Thank you

**Review by asein-sinch — APPROVED** [review]
